p2p/node.go

```diff
@@ -113,10 +117,17 @@ func NewQriNode(r repo.Repo, p2pconf *config.P2P, pub event.Publisher) (node *Qr
 		msgState:    &sync.Map{},
 		pub:         pub,
 		receiversMu: sync.Mutex{},
-		// Make sure we always have proper IOStreams, this can be set
-		// later
+		// Make sure we always have proper IOStreams, this can be set later
 		LocalStreams: ioes.NewDiscardIOStreams(),
 	}
+
+	// TODO(ramfox): when we remove the repo, we need to be sure that the node
+	// has a reference to the dscache in order for it to be possible to resolve
+	// any references locally
+	// use repo and dscache reference inside repo to create a local resolver
+	node.localResolver = dsref.SequentialResolver(r.Dscache(), r)
```

Comment:
Okay, want an opinion here. When I first coded this up, I had the `localResolver` come from a param passed into `NewQriNode`. I remembered we didn't really want other systems, other than the instance, to have control over resolution.

However, we need the qri node to have a local resolver when we get a request from a peer to see if we have the ref in question on our node.

I ended up removing the passed in `localResolver` param, and creating a local resolver when we create the node. Honestly, I think mostly for aesthetic reasons. 

I'm wondering if we should be switching over to a system closer to the way we do NewInstance: adding repo, bus, and localResolver fields to the p2p config, and setting them dynamically through options.

Reply:
> Honestly, I think mostly for aesthetic reasons.

I know the feel. It feels like adding complexity to the `NewQriNode` constructor, but it's _necessary_ complexity. We need to be able to configure the notion of "local" at the instance layer. qri cloud uses a different set of resolvers as it's local stack, making this code a no-go as it's currently written.

I think `NewQriNode` should just take the extra param for now. This is a low level API, where it's supposed to be OK to have very complex constructors.